File: java/src/test/java/com/dddeurope/niuz/fixtures/InMemoryArticles.java
package com.dddeurope.niuz.fixtures;

import com.dddeurope.niuz.newsroom.Article;
import com.dddeurope.niuz.newsroom.ArticleRepository;

import java.util.HashMap;
import java.util.Map;

public class InMemoryArticles implements ArticleRepository {
    final Map<String, Article> articles = new HashMap<>();

    @Override
    public void save(Article article) {
        articles.put(article.getId(), article);
    }

    @Override
    public Article getByArticleId(String articleId) {
        return articles.get(articleId);
    }}

File: dotnet/niuz.application.test/fixtures/InMemoryArticles.cs
using System.Collections.Generic;
using niuz.application.entities;
using niuz.application.repositories;

namespace niuz.application.fixtures
{
    public class InMemoryArticles : IArticleRepository
    {
        private readonly Dictionary<string, Article> articles = new Dictionary<string, Article>();

        public void Save(Article article)
        {
            articles[article.Id] = article;
        }

        public Article GetByArticleId(string articleId)
        {
            return articles[articleId];
        }
    }
}